common/scala/src/main/scala/org/apache/openwhisk/core/connector/Message.scala

```diff
@@ -340,7 +345,22 @@ object Activation extends DefaultJsonProtocol {
       "conductor",
       "memory",
       "causedBy",
-      "size")
+      "size",
+      "actionStatusCode")
+
+  /** Get "StatusCode" from result response **/
+  def getActivationStatusCode(result: Option[JsValue]): Option[Int] = {
+    val statusCode = JsHelpers.getFieldPath(result.get.asJsObject, ERROR_FIELD, "statusCode")
+
+    statusCode match {
+      case Some(value) => Some(value.convertTo[Int])
```

Comment:
Can be simplified  a bit here and also below
```scala
Some(JsNumber(code)) => Some(code.intValue())
```

Reply:
the `statusCode` here is a use defined value - it could be of any type, including integer, string or other legal JSON values. Since this function is trying to interpret a use response it needs to be more defensive and also handle  `case _` when there are no matches.

also note that the controller handles string and int status codes.